compiler/src/securityAnnotations.ml

```diff
@@ -0,0 +1,113 @@
+open Utils
+open Prog
+
+(* Open questions / bikeshedding
+
+   - Is a pair (p, q) fine or should we use the more verbose { n:p, s:q }?
+*)
+
+(* Abstract syntax for security annotations *)
+type simple_level = Public | Secret | Named of Name.t
+
+let named n = Named n
+
+type level = { normal : simple_level; speculative : simple_level }
+
+let diag d = { normal = d; speculative = d }
+let transient = { normal = Public; speculative = Secret }
+
+type typ = Direct of level | Indirect of { ptr : level; value : level }
```

Comment:
I misread it, in fact there is two way of adding annotation "the old one" and the new one.

Reply:
You are perfectly right. Do you think it would also make sense to allow “msf” as a key-word in these signatures?
~~~
#[sct="{ptr: transient, val: secret } * msf -> msf"]
~~~

Reply:
I think so yes.

Reply:
Done.